lib/pure/includes/osenv.nim

```diff
@@ -265,8 +184,36 @@ else:
     ## * `existsEnv proc <#existsEnv,string>`_
     ## * `putEnv proc <#putEnv,string,string>`_
     ## * `delEnv proc <#delEnv,string>`_
-    getEnvVarsC()
-    for i in 0..high(environment):
-      var p = find(environment[i], '=')
-      yield (substr(environment[i], 0, p-1),
-             substr(environment[i], p+1))
+    when defined(windows) and not defined(nimscript):
+      block:
+        when useWinUnicode:
+          var env = getEnvironmentStringsW()
```

Comment:
can you factor using a template like you did earlier to avoid duplication?

Reply:
It turned out that there was differences in behavior I didn't know how to model in a template, like `if eend[1].int == 0: break` vs `if eend[1] == '\0': break`. If this actually is possible, let me know and I'll factor it out.

Reply:
```nim
template impl(zero, ...) =
  ...
  if eend[1] == zero: break
  ...
impl(0, ...)
impl('\0', ...)
```